### PR TITLE
Make user buffer name configurable

### PR DESCRIPTION
**Pull request by erhoo82**
`LayerNormLinear` and `Linear` modules can be used for different layers in a Transformer block. This PR makes the userbuffer name configurable thus enabling the use of these modules to have different communication buffers and identifying tensor-parallel communication configurations of each userbuffer-based communication overlap instance.

### CODE REVIEW

File: transformer_engine/pytorch/module/layernorm_linear.py [hunk 1]
```diff
@@ -670,6 +674,10 @@ def __init__(
         self.ub_bulk_dgrad = ub_bulk_dgrad
         self.ub_split_ag = ub_split_ag
         self.ub_atomic_gemm_ag = ub_atomic_gemm_ag
+        if any([i for i in [ub_bulk_wgrad, ub_bulk_dgrad, ub_split_ag] if i]):
```
Comment (timmoon10) [inline transformer_engine/pytorch/module/layernorm_linear.py:677]:
```suggestion
        if any([ub_bulk_wgrad, ub_bulk_dgrad, ub_split_ag]):
```

File: transformer_engine/pytorch/module/linear.py [hunk 2]
```diff
@@ -574,6 +578,9 @@ def __init__(
         self.ub_split_ag = ub_split_ag
         self.ub_atomic_gemm_rs = ub_atomic_gemm_rs
         self.ub_atomic_gemm_ag = ub_atomic_gemm_ag
+        if any([i for i in [ub_atomic_gemm_rs, ub_atomic_gemm_ag] if i]):
```
Comment (timmoon10) [inline transformer_engine/pytorch/module/linear.py:581]:
```suggestion
        if any([ub_atomic_gemm_rs, ub_atomic_gemm_ag]):
```

**Review by timmoon10 — APPROVED** [review]
Overall LGTM. My suggestions are stylistic.

**Comment by erhoo82** [comment]
Thanks for the fix!

**Comment by erhoo82** [comment]
```
  File "/usr/local/lib/python3.10/dist-packages/transformer_engine/pytorch/module/layernorm_linear.py", line 675
    device: Union[torch.device, str] = "cuda",
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
SyntaxError: duplicate argument 'device' in function definition
```
The above looks to be irrelevant error?

**Comment by erhoo82** [comment]
@ksivaman Can you trigger CI test and help merge this?

File: transformer_engine/pytorch/module/layernorm_linear.py [hunk 3]
```diff
@@ -669,6 +671,8 @@ def __init__(
         ub_bulk_dgrad: bool = False,
         ub_split_ag: bool = False,
         ub_atomic_gemm_ag: bool = False,
+        ub_name: Optional[str] = None,
+        device: Union[torch.device, str] = "cuda",
```
Comment (ksivaman) [inline transformer_engine/pytorch/module/layernorm_linear.py:675]:
```suggestion
```

**Pull request merged**